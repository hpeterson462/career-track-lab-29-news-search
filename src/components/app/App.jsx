import React from 'react';
import Loading from './loading/Loading.jsx';

export default function App() {
  return <Loading />;
}
