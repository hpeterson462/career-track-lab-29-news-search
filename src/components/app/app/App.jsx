import React from 'react';
import Loading from '../Loading/Loading.jsx';

export default function App() {
  return <Loading />;
}
