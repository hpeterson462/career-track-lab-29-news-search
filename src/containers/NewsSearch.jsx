import React, { Component } from 'react';
import Loading from '../components/Loading/Loading';
import AllArticles from '../components/AllArticles/AllArticles';
import { getApi } from '../services/api';

export default class NewsSearch extends Component {
  state = {
    loading: false,
    text: '',
    articles: []
  }

  handleChange = ({ target }) => {
    this.setState({ text: target.value })
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    this.setState({ loading: true })
    getApi(this.state.text)
      .then(articles => this.setState({ articles }))
  }

  render() {
    const { loading, text, articles } = this.state;
    if (loading) return <Loading />;

    return (
      <div>
        {/* <input type="text" onChange={this.handleChange} /> */}
        {/* <Search
          text={text}
          onChange={this.handleChange}
          onSubmit={this.handleSubmit}
        /> */}

        <form onSubmit={this.handleSubmit}>
          <label htmlFor="text">Search News</label>
          <input
            id="text"
            type="text"
            name="text"
            value={text}
            onChange={this.handleChange}
          />
          <button type="submit">Search</button>
        </form>

        <AllArticles
          articles={articles}
        />
      </div>
    );
  }
}
