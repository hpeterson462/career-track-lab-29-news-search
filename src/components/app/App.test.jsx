import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../app/App';
import NewsSearch from '../../containers/NewsSearch';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<NewsSearch />);
    expect(asFragment()).toMatchSnapshot();
  });
});
