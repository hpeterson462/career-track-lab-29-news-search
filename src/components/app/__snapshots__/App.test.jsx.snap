// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <img
    alt="dog carrying newspaper"
    class="Loading"
    src="https://cdn.dribbble.com/users/15084/screenshots/729455/dogloader-dribbble.gif"
  />
</DocumentFragment>
`;
